src/pages/VideoDetail/VideoDetail.page.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect } from 'react';
+import moment from 'moment';
+import { useParams, useHistory } from 'react-router-dom';
+import { useVideoContext } from '../../utils/store/context';
+import Carrusel from '../../components/Carrusel';
+import {
+  Container,
+  ContainerVideo,
+  Video,
+  ContainerDetail,
+  ContainerRelatedVideosList,
+  Title,
+  Description,
+  Date,
+  ChannelTitle,
+} from './VideoDetail.styled';
+
+const VideoDetailPage = () => {
+  const { id } = useParams();
+  const { push } = useHistory();
+  const { videos, video, setVideoDetail } = useVideoContext();
+
+  const load = () => {
+    if (videos.length === 0) {
+      push(`/`);
+    } else {
+      setVideoDetail(id);
+    }
+  };
+
+  useEffect(() => {
+    load();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
```

Comment:
You don't need to define the `load` function, you can put the code in the `useEffect` like this:

```suggestion
  useEffect(() => {
    if (videos.length === 0) {
      push(`/`);
    }

    setVideoDetail(id);
  }, [videos, id]);
```